### PR TITLE
Try to fix flaky tests

### PR DESCRIPTION
**Pull request by tgodzik**

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala [hunk 1]
```diff
@@ -1147,7 +1147,7 @@ class MetalsLanguageServer(
   }
 
   private def didCompileTarget(report: CompileReport): Unit = {
-    if (!isReliableFileWatcher) {
+    if (isReliableFileWatcher) {
```
Comment (dos65) [inline metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala:1150]: Missing not after revertion
```suggestion
    if (!isReliableFileWatcher) {
```

File: tests/unit/src/main/scala/tests/TestingServer.scala [hunk 2]
```diff
@@ -135,8 +135,7 @@ final class TestingServer(
     bspGlobalDirectories = bspGlobalDirectories,
     sh = sh,
     time = time,
-    // relying on the macOS/Windows file watchers causes flaky test failures.
-    isReliableFileWatcher = Properties.isLinux
+    isReliableFileWatcher = System.getenv("CI") == "true"
```
Comment (dos65) [inline tests/unit/src/main/scala/tests/TestingServer.scala:138]:
```suggestion
    isReliableFileWatcher = System.getenv("CI") != "true"
```

Comment (tgodzik) [reply tests/unit/src/main/scala/tests/TestingServer.scala:138]: :man_facepalming: Thanks!

**Review by dos65 — APPROVED** [review]

**Comment by tgodzik** [comment]
They should be a bit less flaky, though there are some more tests on my radar that I need to take a look into.

**Pull request merged**